staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -1069,11 +1069,15 @@ func (s *GenericAPIServer) Eventf(eventType, reason, messageFmt string, args ...
 			Name:      fmt.Sprintf("%v.%x", ref.Name, t.UnixNano()),
 			Namespace: ref.Namespace,
 		},
-		InvolvedObject: ref,
-		Reason:         reason,
-		Message:        fmt.Sprintf(messageFmt, args...),
-		Type:           eventType,
-		Source:         corev1.EventSource{Component: "apiserver", Host: host},
+		InvolvedObject:      ref,
+		Reason:              reason,
+		Message:             fmt.Sprintf(messageFmt, args...),
+		Type:                eventType,
+		Source:              corev1.EventSource{Component: "apiserver", Host: host},
+		EventTime:           t,
+		ReportingController: "apiserver",
+		ReportingInstance:   host,
+		Action:              "shutdown",
```

Comment:
'Shutdown` ?, 'ShuttingDown' ? - not sure there is a convention. I would grep k/k source code for some examples.

Reply:
mhm, `Eventf` method seems to be generic and not tied to "shutdown" events. Maybe we should pass the action as a param ?